geozero/src/csv/csv_error.rs

```diff
@@ -0,0 +1,10 @@
+//! CSV Error type.
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CsvError {
+    #[error("column not found or null")]
```

Comment:
My understanding of @nyurik's request in #138 was to avoid heap allocations.


That doesn't apply to GeozeroError::ColumnNotFound - so I don't see the benefit of adding this CsvError::ColumnNotFound. 

Is there some benefit I'm missing? If not, please revert that portion of the PR and continue to use GeozeroError::ColumnNotFound.

Reply:
I think #138 has two goals, having an error enum per format and making sure those error enums avoid String allocations. 

I don't have strong feelings about the former, and this is a pretty inoffensive change. So in the interest of moving forward, I'm going to merge this as is (after moving the doc comment).